### PR TITLE
Initial formal platform support

### PR DESCRIPTION
**Pull request by lethalbit**
This PR adds initial platform support and modifications needed for integrated formal verification within Torii.

### CODE REVIEW

File: torii/platform/formal.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+	]
+
+	def __init__(self, mode: Literal['bmc', 'prove', 'cover', 'live']) -> None:
+		super().__init__()
+
+		self.mode = mode
+
+	@property
+	def _toolchain_env_var(self) -> str:
+		return super()._toolchain_env_var
```
Comment (dragonmux) [inline torii/platform/formal.py:159]: Does this have to exist, or, can it just be removed so the superclass version is used (as would happen with out this anyway)?

**Review by dragonmux — CHANGES_REQUESTED** [review]
Overall this looks good to us - just a couple of items of feedback/queries and with those taken care of, this looks like it should be good to merge. Nice work!

File: torii/platform/formal.py (hunk 1 above)
Comment (lethalbit) [reply torii/platform/formal.py:159]: Yeah that can be dropped, don't even recall why it was there

**Review by dragonmux — APPROVED** [review]
LGTM for merge once the CI completes.

**Pull request merged**